Hexylbenzene library lai

laitcl opened the pull request:
**Added one training reaction in R_addition_MultipleBond**
Training reaction rate constant added for the following reaction:
1-butyl radical + Styrene --> CCCCC[CH]C1C=CC=CC=1
Rate constant calculated by Lawrence Lai, 2017, CBS-QB3 Level of theory

**Added thermo library named Lai_Hexylbenzene**
Added Library file that contains CBS-QB3 thermo calculations performed by Lawrence Lai in 2017.

Species include:
    Hexylbenzene
    Styrene
    Hexylbenzene Radicals on all aliphatic positions
    n-alkylbenzenes from Toluene to Pentylbenzene
    Alkylbenzyl radicals (radical positioned adjacent to aromatic ring) from
    benzyl radical to pentylbenzyl radical
    Indane
    Ethyltetralin
    C1=CC=C2CCCC2[CH]1
    CCC1CCCC2=CC=C[CH]C21
    1-Butyl radical
    1-Pentyl radical
    CC1=C[CH]C=CC1
    CC1[CH]C=CCC=1
    C[C]1C=CCC=C1
    CC1C=C[CH]C=C1
    CCCCCCC1=C[CH]C=CC1
    CCCCCCC1[CH]C=CCC=1
    CCCCCC[C]1C=CCC=C1
    CCCCCCC1C=C[CH]C=C1

alongd commented:
Since the requested change is on a different repo, we could either test locally or change the travis.yml files to checkout the correct branches (have that in a separate commit so we could delete it before merging)

alongd commented:
should be fixed now by https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1159
Thanks @mliu49 !

mliu49 commented:
Could you add the level of theory to the library entries and make of note of hindered rotors where used?

alongd commented on input/kinetics/families/R_Addition_MultipleBond/training/reactions.py:
```diff
@@ -1233,3 +1233,16 @@
+    index = 76,
+    label = "C4H9-1 + C8H8 <=> C12H17-1",
+    degeneracy = 1,
+    kinetics = Arrhenius(A=(236.006, 'cm^3/(mol*s)'), n=2.7878, Ea=(15.4228, 'kJ/mol'), T0=(1, 'K')),
```
Are there really so many significant figures achieved by CBS-QB3?

alongd commented on input/kinetics/families/R_Addition_MultipleBond/training/reactions.py:
```diff
@@ -1233,3 +1233,16 @@
+Training reaction, kinetics calculated by Lawrence Lai under the CBS-QB3 level of theory, 2016
+More information can be found on pharose ~laitcl/Gaussian/HxBUpdated
+""",
+)
```
Could you add an empty line at the end of the file?

alongd commented on input/thermo/libraries/Lai_Hexylbenzene.py:
```diff
@@ -0,0 +1,1907 @@
+
+name = "Hexylbenzene"
+shortDesc = u""
+longDesc = u"""
```
Is there any helpful description you can add?
Perhaps whether they were all done at the same level of theory?
Or why/how were these species selected?
Is this library part of a paper you're gonna publish? If so, could you give the title & author names?
Were BAC used? If so, from the present CanTherm data, or from the new reference added by @aelong (https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1186), which is still unmerged?
What level of theory was used for FREQ? (If it's also CBS-QB3, it means `iop(7/33=1)` aka Hessian wasn't used)

alongd commented on input/thermo/libraries/Lai_Hexylbenzene.py:
```diff
@@ -0,0 +1,1907 @@
+14 H u0 p0 c0 {6,S}
+15 H u0 p0 c0 {7,S}
+16 H u0 p0 c0 {8,S}
+
```
looks like an unnecessary empty line. more like this below

laitcl commented:
Responses to Alon's comments and extra changes

    1. Now added a very lengthy description to Lai_Hexylbenzene library to
    outline level of theory, and intended publication

    2. All adjacency lists in Lai_Hexylbenzene use benzene bond notation
    instead of single/double notation, known to resolve bugs in
    thermochemistry estimations of radicals.

    3. Added thermochemistry for six polycyclic compounds in
    Lai_Hexylbenzene library. They are:
    C1=CCC2CCCC2=C1
    C1=CC2CCCC2=CC1
    C1C=CC2CCCC2C=1
    C1=CCC2CCCCC2=C1
    C1=CC2CCCCC2=CC1
    C1C=CC2CCCCC2C=1

    4. Added empty line at the end of
    R_Addition_MultipleBond/training/reactions.py

    5. Added a reaction Library with two reactions, named Lai_Hexylbenzene,
    parallel to the thermo Library

    6. Did not change significant figures. Added a note in the long description that describes the 
    uncertainty in CBS-QB3 instead. Might do something in the future to address this.

alongd commented:
@laitcl , looking very good. I appreciate the lengthy comments, I'm sure they'll be very useful to users of these libraries. We should also make a small RMG-Py PR to add you kinetic library to the [documentation](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/documentation/source/users/rmg/database/kinetics.rst#libraries).

alongd commented:
Please also organize/squash the commits before we merge this in.

alongd commented:
Thanks @laitcl and @KEHANG !
last request: could squash together the three thermo library commits?
(a commit name `Added Thermochemistry for four species` isn't helpful w/o context)

laitcl commented:
Squashed.

alongd commented:
@laitcl , could you add the kinetic library here + a short explanation to the documentation on the -Py repository?
https://github.com/ReactionMechanismGenerator/RMG-Py/blob/master/documentation/source/users/rmg/database/kinetics.rst

laitcl commented:
Done
